--- a/third_party/sqlite3.33-src/sqlite3.h
+++ b/third_party/sqlite3.33-src/sqlite3.h
@@ -159,7 +159,7 @@ extern "C" {
 **
 ** See also: [sqlite_version()] and [sqlite_source_id()].
 */
-SQLITE_API SQLITE_EXTERN const char sqlite3_version[];
+
 SQLITE_API const char *sqlite3_libversion(void);
 SQLITE_API const char *sqlite3_sourceid(void);
 SQLITE_API int sqlite3_libversion_number(void);

--- a/third_party/sqlite3.33-src/sqlite3.c
+++ b/third_party/sqlite3.33-src/sqlite3.c
@@ -161109,3 +161109,3 @@ extern "C" {
 */
-SQLITE_API const char *sqlite3_libversion(void){ return sqlite3_version; }
+SQLITE_API const char *sqlite3_libversion(void){ return "3.33.0"; }

 /* IMPLEMENTATION-OF: R-25063-23286 The sqlite3_sourceid() function returns a
